Sources/Build/BuildDelegate.swift

```diff
@@ -237,6 +236,36 @@ public final class BuildExecutionContext {
     }
 }
 
+public protocol PackageStructureDelegate {
+    func packageStructureChanged() -> Bool
+}
+
+final class PackageStructureCommand: CustomLLBuildCommand {
+
+    override func getSignature(_ command: SPMLLBuild.Command) -> [UInt8] {
+        // Include build parameters and process env in the signature.
+        var hash = Data()
+        do {
+            hash += try self.ctx.buildParameters.hash()
```

Comment:
Is hash() new? I recommend using a new name like "stableHash" to indicate how important it is this be stable versus Swift builtin behavior.

Reply:
I'll just bring that code inline here. We don't really need a public method.